wakuv2/waku.go

```diff
@@ -1396,27 +1391,18 @@ func (w *Waku) Start() error {
 				if w.cfg.LightClient {
 					w.lightClientConnectionStatus()
 				}
-			case <-peerCountTimer.C:
-				peerCountTimer.Reset(3 * time.Second)
-				newPeerCount := len(w.node.Host().Network().Peers())
-				if newPeerCount != peerCount && w.onPeerStats != nil {
-					peerCount = newPeerCount
-					// TODO: `IsOnline` is not implemented correctly here, however
-					// this is not a problem because Desktop ignores that value.
-					// This should be fixed to as part of issue
-					// https://github.com/status-im/status-go/issues/4628
-					w.onPeerStats(types.ConnStatus{
-						IsOnline: true,
-						Peers:    FormatPeerStats(w.node),
-					})
-				}
-			case c := <-w.topicHealthStatusChan:
-				w.connStatusMu.Lock()
 
+			case <-w.topicHealthStatusChan:
 				// TODO: https://github.com/status-im/status-go/issues/4628
-				// This code is not using the topic health status correctly.
-				// It assumes we are using a single pubsub topic for now
-				latestConnStatus := formatConnStatus(w.node, c)
+
+			case <-w.connectionNotifChan:
```

Comment:
We can do this for lightClients as well because if you see https://github.com/status-im/status-go/blob/3145ab05ff5a3cbd56c9c8f9db76352d474b464c/wakuv2/waku.go#L1296-L1335 , similar functionality is implemented for lightClients. 
Better to merge these 2 into a function and keep it same. just that in case of lightClient we do invoke `w.filterManager.onConnectionStatusChange` additionally.

Reply:
Done in last commit!